lib/app/ui/driver_profile/driver_profile_page.dart

```diff
@@ -0,0 +1,209 @@
+import 'package:age_calculator/age_calculator.dart';
+import 'package:auto_size_text/auto_size_text.dart';
+import 'package:f1_application/app/component/background/background_bottom.dart';
+import 'package:f1_application/app/component/background/background_top.dart';
+import 'package:f1_application/app/component/loading/full_page_loading.dart';
+import 'package:f1_application/generated/assets.dart';
+import 'package:f1_application/lib/datamanagement/repository/google_image_repository.dart';
+import 'package:f1_application/lib/datamanagement/repository/grid_repository.dart';
+import 'package:f1_application/lib/model/driver.dart';
+import 'package:f1_application/app/ui/driver_profile/component/driver_code.dart';
+import 'package:flag/flag.dart';
+import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+import 'package:f1_application/util/countries.dart' as country;
+
+class DriverProfileScreen extends StatefulWidget {
+  const DriverProfileScreen({Key? key}) : super(key: key);
+
+  @override
+  State<DriverProfileScreen> createState() => _DriverProfileScreenState();
+}
+
+class _DriverProfileScreenState extends State<DriverProfileScreen> {
+  @override
+  void initState() {
+    super.initState();
+    DriverModel? selectedDriver = Provider.of<GridRepository>(context, listen: false).getSelectedDriver;
+    WidgetsBinding.instance.addPostFrameCallback((_) {
+      Provider.of<GoogleImageRepository>(context, listen: false).fetchGoogleImage(selectedDriver!.lastName);

```

Comment:
Itt én nem `GoogleImage` -nek hívnám a ViewModelt amit a képernyő kap, hanem `DriverProfileViewModel` -nek, és úgy építeném fel, hogy konstruktor paraméteren kapja meg a Driver objektumot akire létrehozzuk. Így elkerülhető lenne az, hogy a Google kép letöltésekor félrehasználjuk a működést, plusz a későbbi bővítésekre is jobban fel lenne készítve az osztály